utilities/write_batch_with_index/write_batch_with_index_internal.cc

```diff
@@ -94,9 +94,11 @@ int WriteBatchEntryComparator::operator()(
     return -1;
   }
 
-  if (entry1->offset == WriteBatchIndexEntry::kFlagMin) {
+  if (entry1->offset == WriteBatchIndexEntry::kFlagMin &&
+      entry1->search_key == nullptr) {
     return -1;
-  } else if (entry2->offset == WriteBatchIndexEntry::kFlagMin) {
+  } else if (entry2->offset == WriteBatchIndexEntry::kFlagMin &&
+             entry2->search_key == nullptr) {
```

Comment:
I am not sure why this fixes the bug? It actually prevents it from doing any comparison of the key contents. Is that the right algorithm for comparison?